leftwm-core/src/handlers/command_handler.rs

```diff
@@ -143,6 +143,38 @@ fn toggle_state(state: &mut State, window_state: WindowState) -> Option<bool> {
     let window = state.focus_manager.window(&state.windows)?;
     let handle = window.handle;
     let toggle_to = !window.has_state(&window_state);
+    let tag_id = state.focus_manager.tag(0)?;
+
+    let for_active_workspace = |x: &Window| -> bool { x.tag == Some(tag_id) && x.is_managed() };
+
+    if window_state == WindowState::Fullscreen {
+        //Going fullscreen, so we save the window order
+        //or else, we restore it!
+        if toggle_to {
+            let mut to_reorder = helpers::vec_extract(&mut state.windows, for_active_workspace);
+            let handles = to_reorder
+                .iter()
+                .map(|each_handle| each_handle.handle)
+                .collect::<Vec<_>>();
+
+            state.window_order = handles;
+            state.windows.append(&mut to_reorder);
+        } else {
+            let to_reorder = &mut state.windows;
+            let mut window_restored = vec![];
```

Comment:
Would it be better to make this plural (`windows_restores`) nice it is a `Vec`?

Reply:
The `focus_manager` uses a field `pub window_history: VecDeque<MaybeWindowHandle>`, of course in this case "history" is not precisely accurate.

(On a side note, there is a focusing issue that is kind of similar to this one: when a pop-up floating window is opened end "destroyed" e.g. some password dialog, the focus returns to the `main window` an not to the last focused one. If you were interested to extend the PR with this, that would be welcome for sure :wink:)